lib/rules/date.js

```diff
@@ -0,0 +1,33 @@
+import SyncRule from '../types/syncRule';
+
+// eslint-disable-next-line
+const strictISO = /^(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+([+-][0-2]\d:[0-5]\d|Z))|(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d([+-][0-2]\d:[0-5]\d|Z))|(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d([+-][0-2]\d:[0-5]\d|Z))$/;
+
+class DateValidator extends SyncRule {
+    coerce(value) {
+        if (typeof value === 'string' && strictISO.test(value)) {
```

Comment:
Not in our API, but it's totally feasible that we (or someone else, this is OSS) might want to use another format or consume one from a source we don't control.